src/modules/interface.js

```diff
@@ -0,0 +1,231 @@
+import ToDo from './todo.js';
+import Storage from './storage.js';
+
+export default class Interface {
+  static toDoList = Storage.getTasks();
+
+  static completedTasks = [];
+
+  static editList = '';
+
+  static displayToDoList = () => {
+    const toDoListContainer = document.querySelector(
+      '.toDoItem',
+    );
+    const clearBtn = document.querySelector('.clearTask');
+    if (Interface.toDoList.length === 0) {
+      clearBtn.classList.add('hide');
+      toDoListContainer.innerHTML = '';
+      toDoListContainer.innerHTML = 'Add A Task At Least!';
+      toDoListContainer.classList.add('errorMsg');
+    } else {
+      toDoListContainer.innerHTML = '';
+      clearBtn.classList.remove('hide');
+      // sort the ToDo array by index
+      Interface.toDoList.sort((a, b) => a.index - b.index);
+      Interface.toDoList.forEach((task) => {
+        const taskIndex = task.index;
+        const taskDescription = task.description;
+        const listItem = document.createElement('div');
+        listItem.className = 'toDoList';
+        listItem.innerHTML = `
+  <input type="checkbox" class="checkbox" id="${taskIndex}" name="${taskDescription}" value="${taskDescription}"><i id="${taskIndex}" class="fa fa-check hide tick" aria-hidden="true"></i><div class='taskDescription'>${taskDescription}</div>
+  <input type="text" class="editInput hide" id="${taskIndex}" name="${taskDescription}" value="${taskDescription}">
+  <button type="button" class="optionBtn" ><i class="fa fa-ellipsis-v fa-2xl menu" id="${taskIndex}" aria-hidden="true"></i></button>
+  <button type="button" class="deleteBtn hide" ><i class="fa fa-trash fa-xl trash-bin" aria-hidden="true"></i></button>
+  `;
+        toDoListContainer.appendChild(listItem);
+      });
+      // add event listener to each ellipses
+      const optionBtn = document.querySelectorAll('.optionBtn');
+      optionBtn.forEach((btn) => {
+        btn.addEventListener('click', (event) => {
+          const listItem = event.target.closest('.toDoList');
+          if (Interface.editList === '') {
+            Interface.editList = listItem;
+            Interface.editTask(Interface.editList);
+          } else if (Interface.editList) {
+            if (Interface.editList !== listItem) {
+              Interface.cancelEditTask(Interface.editList);
+              Interface.editList = listItem;
+              Interface.editTask(Interface.editList);
+            } else if (Interface.editList === listItem) {
+              Interface.cancelEditTask(listItem);
+              Interface.editTask(Interface.editList);
+            }
+          }
+        });
+      });
+
+      // add event listener to each checkbox
+      const checkboxes = document.querySelectorAll('.checkbox');
+      checkboxes.forEach((checkbox) => {
+        checkbox.addEventListener('change', (event) => {
+          Interface.lineOnTask(event.target);
+        });
+      });
```

Comment:
- [ ] **_This 👇🏾  issue has not been fixed:_** 
> As specified in the project requirements, kindly make sure to update the completed upon user action. The initial state of a task is false but upon user action, it should be changed to true. Please note that the change should also be updated in the localStorage as well as it's currently not. This way when users refresh their browser page they still get the updated state of their tasks.


```javascript

// add event listener to each checkbox
      const checkboxes = document.querySelectorAll('.checkbox');
      checkboxes.forEach((checkbox) => {
        checkbox.addEventListener('change', (event) => {
          Interface.lineOnTask(event.target);
          // Update the local storage here after the checkbox is checked
          const task = Interface.toDoList.find(
            (task) => task.index === parseInt(event.target.id, 10),
          );
          task.done = event.target.checked;          
          Storage.updateTasks(Interface.toDoList);
        });
      });

```

Above is my suggestion for where that state update can happen. After the _**completed line**_ on the task is added or removed, the local storage should be updated.

❗ ❕  **-> The code above has not been tested to work yet, you will use the above snippet as a guide.**